File: C99/백준/Bronze/2798. 블랙잭/블랙잭.c
#include<stdio.h>
int main(void)
{
	int m, n, sum = 0, max = 0;
	int a[100];
	scanf("%d %d", &n, &m);
	for (int i = 0; i < n; i++)
	{
		scanf("%d", &a[i]);
	}
	for (int i = 0; i < n; i++)
	{
		for (int j = i + 1; j < n; j++)
		{
			for (int k = j + 1; k < n; k++)
			{
				sum = a[i] + a[j] + a[k];
				if (sum >= max && sum <= m)
				{
					max = sum;
				}
			}
		}
	}
	printf("%d", max);

	return 0;
}

File: Python/백준/Bronze/2798. 블랙잭/블랙잭.py
import sys
n, m = map(int, sys.stdin.readline().split())

list = list(map(int, sys.stdin.readline().split()))
    
max = 0
for i in range(n-2):
    for j in range(i+1,n-1):
        sum = 0
        for k in range(j+1,n):
            sum = list[i] + list[j] + list[k]
            if(sum <= m and max < sum):
                max = sum

print(max)

